leaflet-image.js

```diff
@@ -204,20 +213,76 @@ module.exports = function leafletImage(map, callback) {
         im.crossOrigin = '';
 
         im.onload = function() {
+            if (loaded) return;
+            loaded = true;
             ctx.drawImage(this, x, y, size[0], size[1]);
             callback(null, {
                 canvas: canvas
             });
         };
 
-        im.src = url;
+        if (useAjax && !isBase64) {
+            if (cache === undefined) {
+                getBase64(url, function (data) {
+                    im.src = imageCache[marker._icon.src] = 'data: image/png;base64, ' + data;
+                    if (!loaded) {
+                        setTimeout(function() {
+                            im.onload();
+                        },0);
+                    }
+                });
+            } else {
+                im.src = cache;
+                if (!loaded) {
+                    setTimeout(function() {
+                        im.onload();
```

Comment:
The image isn't yet added to the page, but is base64.  `setTimeout(fn, 0)` ensures it is loaded when the next available context is, which (since we just set the src) is just after the src is rendered.  Otherwise the image is not yet rendered, thus resulting in a blank image.